### PR TITLE
ENT-3164: Update dependencies for 3.11 Beta

### PR DESCRIPTION
**Pull request by Lex-2008**

### CODE REVIEW

**Comment by jimis** [comment]
Please add a "changelog" tag to the removed deps commit message.

File: deps-packaging/lcov/cfbuild-lcov.spec [hunk 1]
```diff
@@ -24,7 +24,7 @@ exit 0
 
 %install
 rm -rf $RPM_BUILD_ROOT
-make install PREFIX=$RPM_BUILD_ROOT
+make install DESTDIR=$RPM_BUILD_ROOT PREFIX=/usr CFG_DIR=/etc
```
Comment (jimis) [inline deps-packaging/lcov/cfbuild-lcov.spec]: Why was this change needed? Add the reason to the commit message.

Comment (Lex-2008) [inline deps-packaging/lcov/cfbuild-lcov.spec]: "Copy `make install` command from build script provided in the package" sounds good?

Also, I squashed together commits for changing version and making necessary changes in our buildscripts. Do you want me to unsquash them back?

File: build-scripts/compile-options [hunk 2]
```diff
@@ -3,7 +3,6 @@
 # Calculate dependencies and build parameters according to the environment.
 # SOURCE this file, don't execute.
 
-
```
Comment (jimis) [inline build-scripts/compile-options]: Can you add an explanation of why we removed the dependency in the commit message? (It doesn't have to be in the changelog.) For reference in case somebody tries to bring back the dependency in the future. Also mention the ticket number.

File: build-scripts/compile-options [hunk 3]
```diff
@@ -171,7 +171,7 @@ case "$ROLE" in
         ;;
     # AGENT-ONLY dependencies
     agent)
-        var_append DEPS "libcurl postgresql"
```
Comment (jimis) [inline build-scripts/compile-options]: Reason for removal in the commit message, and mention the ticket number.

File: deps-packaging/lcov/cfbuild-lcov.spec (hunk 1 above)
Comment (jimis) [inline deps-packaging/lcov/cfbuild-lcov.spec]: Explanation needs to show the root cause, like "upstream changed build instructions". No need to unsquash.

**Comment by jimis** [comment]
Nice, thank you @Lex-2008 !

What do you think is best? Cherry-picking the relevant commits to 3.10.x/3.7.x now, or waiting for the release of these branches and updating dependencies then?

**Comment by Lex-2008** [comment]
I think we do only minor version updates on that branches, so, for example, 3.10 will always be on git 2.10.x branch, while 3.11 is on git 2.13.0 now, after this commit. Hence, these commits are not relevant to those branches.

**Comment by jimis** [comment]
Almost. But while we try to keep breakage to a minimum in LTS branches, we would still upgrade the major version of a dependency if deemed important, if for example the old version of the dependency imposed a security risk.

**Comment by Lex-2008** [comment]
Hm, then we need to investigate it on a case-by-case basis. Let's have a look at it when we come to upgrading relevant 3.7/3.11 branches

**Comment by jimis** [comment]
On Tue, May 16, 2017 at 11:26 AM, Aleksei Shpakovskii <
notifications@github.com> wrote:

> Hm, then we need to investigate it on a case-by-case basis. Let's have a
> look at it when we come to upgrading relevant 3.7/3.11 branches


OK!
